controller/linodeplacementgroup_controller.go

```diff
@@ -0,0 +1,342 @@
+//
+//nolint:dupl // this is same as VPC, worth making generic later.
+func (r *LinodePlacementGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultedLoopTimeout(r.ReconcileTimeout))
+	defer cancel()
+
+	log := ctrl.LoggerFrom(ctx).WithName("LinodePlacementGroupReconciler").WithValues("name", req.NamespacedName.String())
+
+	linodeplacementgroup := &infrav1alpha1.LinodePlacementGroup{}
+	if err := r.Client.Get(ctx, req.NamespacedName, linodeplacementgroup); err != nil {
+		if err = client.IgnoreNotFound(err); err != nil {
+			log.Error(err, "Failed to fetch LinodePlacementGroup")
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	pgScope, err := scope.NewPlacementGroupScope(
+		ctx,
+		r.LinodeApiKey,
+		scope.PlacementGroupScopeParams{
+			Client:               r.Client,
+			LinodePlacementGroup: linodeplacementgroup,
+		},
+	)
+	if err != nil {
+		log.Error(err, "Failed to create Placement Group scope")
+
+		return ctrl.Result{}, fmt.Errorf("failed to create Placement Group scope: %w", err)
+	}
+
+	return r.reconcile(ctx, log, pgScope)
+}
+
+func (r *LinodePlacementGroupReconciler) reconcile(
+	ctx context.Context,
+	logger logr.Logger,
+	pgScope *scope.PlacementGroupScope,
+) (res ctrl.Result, err error) {
+	res = ctrl.Result{}
+
+	pgScope.LinodePlacementGroup.Status.Ready = false
+	pgScope.LinodePlacementGroup.Status.FailureReason = nil
+	pgScope.LinodePlacementGroup.Status.FailureMessage = util.Pointer("")
+
+	failureReason := infrav1alpha1.LinodePlacementGroupStatusError("UnknownError")
+	//nolint:dupl // Code duplication is simplicity in this case.
+	defer func() {
+		if err != nil {
+			pgScope.LinodePlacementGroup.Status.FailureReason = util.Pointer(failureReason)
+			pgScope.LinodePlacementGroup.Status.FailureMessage = util.Pointer(err.Error())
+
+			conditions.MarkFalse(pgScope.LinodePlacementGroup, clusterv1.ReadyCondition, string(failureReason), clusterv1.ConditionSeverityError, err.Error())
+
+			r.Recorder.Event(pgScope.LinodePlacementGroup, corev1.EventTypeWarning, string(failureReason), err.Error())
+		}
+
+		// Always close the scope when exiting this function so we can persist any LinodePlacement Group changes.
+		// This ignores any resource not found errors when reconciling deletions.
+		if patchErr := pgScope.Close(ctx); patchErr != nil && utilerrors.FilterOut(util.UnwrapError(patchErr), apierrors.IsNotFound) != nil {
+			logger.Error(patchErr, "failed to patch LinodePlacementGroup")
+
+			err = errors.Join(err, patchErr)
+		}
+	}()
+
+	// Delete
+	if !pgScope.LinodePlacementGroup.ObjectMeta.DeletionTimestamp.IsZero() {
+		failureReason = infrav1alpha1.DeletePlacementGroupError
+
+		res, err = r.reconcileDelete(ctx, logger, pgScope)
+
+		return
+	}
+
+	// Add the finalizer if not already there
+	err = pgScope.AddFinalizer(ctx)
+	if err != nil {
+		logger.Error(err, "Failed to add finalizer")
+
+		return
+	}
+
+	// Update
```

Comment:
Since all the relevant fields are immutable I think we can skip handling update for now?

Reply:
Yeah - good point - will remove it